swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/work/WorkProcessor.kt

```diff
@@ -76,31 +79,35 @@ class WorkProcessor(
   }
 
   private fun process(work: WorkItem) {
-    try {
-      resourceTypeHandlers.find {
-        it.handles(work.workConfiguration)
-      }?.let { handler ->
-        log.debug("Processing: $work")
-        when (work.action) {
-          Action.MARK -> handler.mark(work.workConfiguration)
-          Action.NOTIFY -> handler.notify(work.workConfiguration)
-          Action.DELETE -> handler.delete(work.workConfiguration)
-          else -> log.warn("Unknown action {}", work.action.name)
-        }
+    val elapsedTimeMillis = measureTimeMillis {
+      try {
+        resourceTypeHandlers.find {
+          it.handles(work.workConfiguration)
+        }?.let { handler ->
+          log.debug("${handler.javaClass.simpleName} processing: {}", work.workConfiguration.namespace)
+          when (work.action) {
+            Action.MARK -> handler.mark(work.workConfiguration)
+            Action.NOTIFY -> handler.notify(work.workConfiguration)
+            Action.DELETE -> handler.delete(work.workConfiguration)
+            else -> log.warn("Unknown action {}", work.action.name)
+          }
 
-        work.track(success = true)
+          track(work, success = true)
+        }
+      } catch (e: Exception) {
+        log.error("Failed to process: {}", work, e)
+        track(work, success = false)
       }
-    } catch (e: Exception) {
-      log.error("Failed to process: {}", work, e)
-      work.track(success = false)
     }
+
+    registry.timer(workDurationId).record(elapsedTimeMillis, TimeUnit.MILLISECONDS)
```

Comment:
Prefer using the timer here with the advantage that all actions are now accounted for. We used to just measure mark duration.